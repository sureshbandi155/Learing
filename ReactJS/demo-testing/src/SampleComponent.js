import React from 'react';

const SampleComponent = () => (
    <>
        <button>Toggle text</button>
        <div>Sample Text</div>
        {/* <div>Sample Text</div> */}
    </>
);

export default SampleComponent;