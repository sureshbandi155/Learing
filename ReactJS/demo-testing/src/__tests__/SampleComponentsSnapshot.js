import React from 'react';
import { configure, shallow } from 'enzyme';
import ReactSixteenAdapter from 'enzyme-adapter-react-16';
import SampleComponent from '../SampleComponent';
import toJson from 'enzyme-to-json';

configure({ adapter: new ReactSixteenAdapter() })

describe('Testing sampleComponent with snapshot', () => {
    it('renders correctly enzyme', () => {
        const wrapper = shallow(<SampleComponent />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
})
