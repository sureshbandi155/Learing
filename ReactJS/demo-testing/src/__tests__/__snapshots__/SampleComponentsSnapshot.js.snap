// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing sampleComponent with snapshot renders correctly enzyme 1`] = `
<Fragment>
  <button>
    Toggle text
  </button>
  <div>
    Sample Text
  </div>
</Fragment>
`;
